### ISSUE TITLE
cannot send to P2TR (trezor model T)

### ISSUE THREAD

**Issue by mbio16** [sparrowwallet/sparrow#400]
2022-01-16 10:14:39,767 ERROR [JavaFX Application Thread] c.s.s.c.DevicePane [null:-1] Signing Error: ProcessError: Invalid address witness program
com.sparrowwallet.sparrow.io.SignTransactionException: ProcessError: Invalid address witness program
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.io.Hwi.signPSBT(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.io.Hwi$SignPSBTService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@1.5.5/com.sparrowwallet.sparrow.io.Hwi$SignPSBTService$1.call(Unknown Source)
	at javafx.graphics/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

**Comment by mbio16** [comment]
bc1q... -> (bc1q... and bc1p...) leads to this type of error.

**Comment by craigraw** [comment]
Duplicate of #376 - Sparrow is waiting on support for P2TR from the upstream HWI library. Hopefully soon.

**Comment by Giszmo** [comment]
For what it's worth, Sparrow with Trezor One gave me a cryptic error message, too. I think, Sparrow should hint the user at the possibility of the HW not being updated instead of showing cryptic errors.

**Comment by craigraw** [comment]
I think you underestimate the difficulty of understanding every error message for every hardware wallet vendor, all of which are regularly releasing new firmware with new errors without any standardisation in this messaging, or even communication protocol :)
